### ISSUE TITLE
[amp-sidebar] - Customizing amp-sidebar mask

### ISSUE THREAD

**Issue by WillDonohoe** [ampproject/amphtml#10380]
## What's the issue?

In our amp template design the sidebar mask is fully transparent. However adding custom styles to this element causes the validator to fail as it's targeting `i-amphtml-sidebar-mask`. 

It would be useful to customize the mask such as changing the background colour, the opacity and other properties like margins.

A simple solution to this would be to add an `amp-sidebar-mask` class to it to allow developers to add styles or override default styles.

Thanks

**Comment by honeybadgerdontcare** [comment]
@WillDonohoe That is working as intended. Any `i-amphtml-` css class will be prohibited as that's an internal css class for amp runtime. There may be another way to target the mask, adding @camelburrito for that possibility.

**Comment by jridgewell** [comment]
/to @camelburrito. Maybe @tiendt can take this?

**Comment by camelburrito** [comment]
yes @tiendt - do you wanna take this?

**Comment by tiendt** [comment]
@camelburrito yeah sure!

**Comment by kazmenko** [comment]
I found simple solution for this issue. It not causes the validator to fail.
Add to your style:
```
footer.center + div{
   opacity: 0.8;
} 
```
Where footer.center is a last element in <body>.

**Comment by camelburrito** [comment]
This might not work as you are relying on the mask being the last element. Other components or the runtime could change this behavior

**Comment by josegustavo** [comment]
other solution is:

```
[class*="amphtml-sidebar-mask"] {
    background: none;
}
```

**Comment by tylertrotter** [comment]
@josegustavo I came to the same solution and ended up here trying to find out if there are any negative consequences to doing it. It seems like "cheating." As far as you know, is this okay to do?

**Comment by VMBindraban** [comment]
I ended up using this:

```css
body ~ div:last-child {
  background: none;
}
```

I prefer not targeting the mask directly since it can be changed without notice.

**Comment by jridgewell** [comment]
ping @ampproject/wg-ui-and-a11y

**Comment by cathyxz** [comment]
We're doing some work (polish + upgrade) on sidebar that will likely resolve this, the current target launch date is mid November. We're having a design review on the new version of sidebar tomorrow (links incoming). 

/cc @nainar @leafsy

**Comment by kristoferbaxter** [comment]
This has now been addressed with #31999. The mask is a styleable using the `amp-sidebar-mask` classname.
